show_client/interface_cli.go

```diff
@@ -386,215 +386,284 @@ func getInterfaceFecStatus(options sdc.OptionMap) ([]byte, error) {
+	}
+
+	// Exclude LAG members from standalone ports
+	var ports []string
+	for port := range portTbl {
+		if !IsInterfaceInPortchannel(portChannelMemberTbl, port) {
+			ports = append(ports, port)
+		}
+	}
+	var portchannels []string
+	for pc := range portChannelTbl {
+		portchannels = append(portchannels, pc)
+	}
+	keys := append(ports, portchannels...)
+	keys = natsortInterfaces(keys)
+
+	// Optionally filter by interface
+	if intf != "" {
+		found := false
+		for _, k := range keys {
+			if k == intf {
+				found = true
+				keys = []string{intf}
+				break
+			}
+		}
+		if !found {
+			return nil, fmt.Errorf("Got unexpected extra argument %s", intf)
+		}
+	}
+
+	// Emit switchportStatus
+	switchportStatus := make([]map[string]string, 0, len(keys))
+	for _, k := range keys {
+		mode := GetInterfaceSwitchportMode(portTbl, portChannelTbl, vlanMemberTbl, k)
+
+		switchportStatus = append(switchportStatus, map[string]string{
+			"Interface": GetInterfaceNameForDisplay(k),
+			"Mode":      mode,
+		})
+	}
+
+	return json.Marshal(switchportStatus)
 }
 
 // IsInterfaceInPortchannel reports whether interfaceName is a member of any portchannel.
 func IsInterfaceInPortchannel(portchannelMemberTable map[string]interface{}, interfaceName string) bool {
-    if portchannelMemberTable == nil || interfaceName == "" {
-        return false
-    }
-    for k := range portchannelMemberTable {
-        _, member, ok := SplitCompositeKey(k)
-    if ok && member == interfaceName {
-            return true
-        }
-    }
-    return false
-}
\ No newline at end of file
+	if portchannelMemberTable == nil || interfaceName == "" {
+		return false
+	}
+	for k := range portchannelMemberTable {
+		_, member, ok := SplitCompositeKey(k)
+		if ok && member == interfaceName {
+			return true
+		}
+	}
+	return false
+}
+
+// 'expected' subcommand ("show interface neighbor expected")
+// admin@sonic: redis-cli -n 4 HGETALL 'DEVICE_NEIGHBOR|Ethernet2'
+// 1) "name"
+// 2) "DEVICE01T1"
+// 3) "port"
+// 4) "Ethernet1"
+//
+// admin@sonic: redis-cli -n 4 HGETALL "DEVICE_NEIGHBOR_METADATA|DEVICE01T1"
+// 1) "hwsku"
+// 2) "Arista-VM"
+// 3) "mgmt_addr"
+// 4) "0.0.0.0"
+// 5) "type"
+// 6) "BackEndLeafRouter"
+func getInterfaceNeighborExpected(options sdc.OptionMap) ([]byte, error) {
+	// Fetch DEVICE_NEIGHBOR
+	neighborTbl, err := GetMapFromQueries([][]string{{"CONFIG_DB", "DEVICE_NEIGHBOR"}})
+	if err != nil {
+		log.Errorf("Failed to get DEVICE_NEIGHBOR: %v", err)
+		return nil, err
+	}
+
+	// Fetch DEVICE_NEIGHBOR_METADATA
+	metaTbl, err := GetMapFromQueries([][]string{{"CONFIG_DB", "DEVICE_NEIGHBOR_METADATA"}})
+	if err != nil {
+		log.Errorf("Failed to get DEVICE_NEIGHBOR_METADATA: %v", err)
+		return nil, err
+	}
+
+	out := make(map[string]map[string]string)
+	for localIf := range neighborTbl {
+		device := GetFieldValueString(neighborTbl, localIf, "", "name")
+		if device == "" {
+			continue
+		}
+		remotePort := GetFieldValueString(neighborTbl, localIf, "", "port")
```

Comment:
Do we need the logic for checking if port is "", then we should reassign it to None or is port allowed to be an empty string if its missing?